plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/data/LayoutMetaDataService.java

```diff
@@ -307,17 +288,47 @@ public void register(final LayoutAlgorithmData algorithmData) {
 
         @Override
         public void register(final LayoutOptionData optionData) {
+            
+            // #1 register fully qualified id
             String id = optionData.getId();
             layoutOptionMap.put(id, optionData);
 
-            // register as suffix (which include groups)
+            // #2 register allowed suffixes (this may include every group)
+            // Example: allowed suffixes for 'org.eclipse.elk.foo.bar.option' are 
+            //  --> foo.bar.option
+            //  --> bar.option
+            //  --> option
             String suffix = id.substring(id.lastIndexOf('.') + 1, id.length());
```

Comment:
This won't work with option IDs that only consist of a single component, will it?